From facb67e3e6575b9954efee7d94e4839d57cc4422 Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Wed, 16 Jul 2014 20:42:11 -0400
Subject: [md] dm: remove symbol export for dm_set_device_limits

Message-id: <1405543364-20901-8-git-send-email-snitzer@redhat.com>
Patchwork-id: 85890
O-Subject: [RHEL7.1 PATCH 07/40] dm: remove symbol export for dm_set_device_limits
Bugzilla: 1117872
RH-Acked-by: Mikulas Patocka <mpatocka@redhat.com>
RH-Acked-by: Heinz Mauelshagen <heinzm@redhat.com>

BZ: 1117872

Upstream commit 11f0431be2f99c574a65c6dfc0ca205511500f29
Author: Mike Snitzer <snitzer@redhat.com>
Date:   Tue Jun 3 10:30:28 2014 -0400

    dm: remove symbol export for dm_set_device_limits

    There is no need for code other than DM core to use dm_set_device_limits
    so remove its EXPORT_SYMBOL_GPL.  Also, cleanup a couple whitespace nits.

    Signed-off-by: Mike Snitzer <snitzer@redhat.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/md/dm-table.c b/drivers/md/dm-table.c
index bc443d1..f10831b 100644
--- a/drivers/md/dm-table.c
+++ b/drivers/md/dm-table.c
@@ -465,8 +465,8 @@ int dm_get_device(struct dm_target *ti, const char *path, fmode_t mode,
 }
 EXPORT_SYMBOL(dm_get_device);
 
-int dm_set_device_limits(struct dm_target *ti, struct dm_dev *dev,
-    sector_t start, sector_t len, void *data)
+static int dm_set_device_limits(struct dm_target *ti, struct dm_dev *dev,
+    sector_t start, sector_t len, void *data)
 {
  struct queue_limits *limits = data;
  struct block_device *bdev = dev->bdev;
@@ -499,7 +499,6 @@ int dm_set_device_limits(struct dm_target *ti, struct dm_dev *dev,
        (unsigned int) (PAGE_SIZE >> 9));
  return 0;
 }
-EXPORT_SYMBOL_GPL(dm_set_device_limits);
 
 /*
  * Decrement a device's use count and remove it if necessary.
diff --git a/drivers/md/dm.c b/drivers/md/dm.c
index 91cf20d..c8715f0 100644
--- a/drivers/md/dm.c
+++ b/drivers/md/dm.c
@@ -1583,7 +1583,6 @@ static int dm_merge_bvec(struct request_queue *q,
   * just one page.
   */
  else if (queue_max_hw_sectors(q) <= PAGE_SIZE >> 9)
-
   max_size = 0;
 
 out:
diff --git a/include/linux/device-mapper.h b/include/linux/device-mapper.h
index 5eeeab4..ecda6e8 100644
--- a/include/linux/device-mapper.h
+++ b/include/linux/device-mapper.h
@@ -115,12 +115,6 @@ typedef int (*dm_busy_fn) (struct dm_target *ti);
 
 void dm_error(const char *message);
 
-/*
- * Combine device limits.
- */
-int dm_set_device_limits(struct dm_target *ti, struct dm_dev *dev,
-    sector_t start, sector_t len, void *data);
-
 struct dm_dev {
  struct block_device *bdev;
  fmode_t mode;
@@ -132,7 +126,7 @@ struct dm_dev {
  * are opened/closed correctly.
  */
 int dm_get_device(struct dm_target *ti, const char *path, fmode_t mode,
-       struct dm_dev **result);
+    struct dm_dev **result);
 void dm_put_device(struct dm_target *ti, struct dm_dev *d);
 
 /*
-- 
1.7.1